Library/Homebrew/formula.rb

```diff
@@ -2393,8 +2292,87 @@ def to_hash
   end
 
   # @private
-  def to_hash_with_variations
-    hash = to_hash
+  def to_api_hash
+    api_hash = {
+      "desc"                 => desc,
+      "license"              => SPDX.license_expression_to_string(license),
+      "homepage"             => homepage,
+      "urls"                 => urls_hash.transform_values(&:compact),
+      "post_install_defined" => post_install_defined?,
+      "ruby_source_path"     => ruby_source_path,
+      "ruby_source_checksum" => { "sha256" => ruby_source_checksum&.hexdigest },
```

Comment:
I believe this was added by @Bo98 on purpose to potentially allow for different hashing algorithms in the future. Not sure if we still need that flexibility though.

Reply:
It's good to design things in a generic fashion as it makes things so much easier down the line (though obviously not if it increases complexity greatly). With that said consistency should be the first priority here so we should consistently do one way or consistently the other way.

Like how we have a Generic OS that realistically isn't actually used, but makes things much easier down the line should we need.

Reply:
I guess if we move to a new hashing algorithm it wouldn't be hard to add a new key but since that's not something we plan on doing anytime soon I'll update it to match @MikeMcQuaid's suggestion.

Reply:
Addressed in https://github.com/Homebrew/brew/pull/16541/commits/3a70b6aeae363c66a6d35f8d51397350aacae35a